Fix install script

bajo opened the pull request:
This introduces a script, SetupSquirrelUser.sh, that makes sure that the squirrel user is created and that the sudo group is able to run commands with sudo. This is not enabled on the Festo robotino image. 

Further, the InstallSquirrel.sh makes sure to be run as squirrel user, provides the installation of ROS and further dependencies. 
Right now we need to run it 3 times to make the basic, the ros and the upstart installation. I will change this later, because I am running out of time right now. Sorry. 

Readme will be changed later as well.
For now: 
1.) run SetupSquirrelUser.sh as robotino
2.) InstallSquirrel.sh as squirrel (as often as needed. up to 3 times)

bajo committed:
Adapt Install script. Add a pre-install script

bajo committed:
Update SetupSquirrelUser.sh

bajo committed:
Update SetupSquirrelUser.sh

bajo commented:
@ipa-nhg I will send an update to this pr in the next hour or so. Please wait with the review until then. 
Thanks.

bajo committed:
Multiply improvements

- add check for correct user
- add check for sudo
- add check for groups
- add check for sudoers file
- Create squirrel user

bajo committed:
Update README.md

Hard coded links will be updated as soon as this has been merged into upstream

bajo commented:
@ipa-nhg PR should be ready for review now. I tested it multiple times on tuw-robotino2. 
Should be woking fine on all robots now.

NadiaHG commented on setup_robotino/InstallSquirrel.sh:
```diff
@@ -11,17 +11,96 @@ EOF
+#### FUNCTION ROS INSTALLATION
+function RosInstallation {
+  echo -e "\n${green}INFO: Setup ROS${NC}\n"
+  sudo sh -c '. /etc/lsb-release && echo "deb http://packages.ros.org.ros.informatik.uni-freiburg.de/ros/ubuntu $DISTRIB_CODENAME main" > /etc/apt/sources.list.d/ros-latest.list'
```
why are you using this mirror and not the official one?

```
  echo -e "\n${green}   INFO: Setup your source.list${NC}\n"
  sudo sh -c 'echo "deb http://packages.ros.org/ros/ubuntu trusty main" > /etc/apt/sources.list.d/ros-latest.list'
  echo -e "\n${green}   INFO: Set up your keys${NC}\n"
  wget http://packages.ros.org/ros.key -O - | sudo apt-key add -
```

bajo replied:
This mirror is a lot faster than the official server. Also it is mentioned on the ROS installation webpage as well. http://wiki.ros.org/ROS/Installation/UbuntuMirrors

NadiaHG commented on setup_robotino/InstallSquirrel.sh:
```diff
@@ -11,17 +11,96 @@ EOF
+  sudo apt-key adv --keyserver hkp://ha.pool.sks-keyservers.net:80 --recv-key 0xB01FA116
+  sudo apt-get update
+  sudo apt-get install ros-indigo-ros-base
+  sudo rosdep init
```
missed: `rosdep update`

bajo replied:
rosdep update is later called in the SquirrelSetup.sh where it is executed as squirrel. If I put it in here it will be as robotino.

NadiaHG commented on setup_robotino/InstallSquirrel.sh:
```diff
@@ -11,17 +11,96 @@ EOF
+  sudo sh -c '. /etc/lsb-release && echo "deb http://packages.ros.org.ros.informatik.uni-freiburg.de/ros/ubuntu $DISTRIB_CODENAME main" > /etc/apt/sources.list.d/ros-latest.list'
+  sudo apt-key adv --keyserver hkp://ha.pool.sks-keyservers.net:80 --recv-key 0xB01FA116
+  sudo apt-get update
+  sudo apt-get install ros-indigo-ros-base
```
why base and not desktop version? you will also need `ros-indigo-rosbash and python-ros-*`

add the arguments  `-y --force-yes` to apt-get install

bajo replied:
I am not sure if we need all packages that are provided by desktop or desktop-full. 
I was hoping that most dependencies will be resolved during the first rosdep install. 
How shall I proceed here?

I will add those arguments.

NadiaHG commented:
:+1:  thanks @bajo

bajo committed:
Update InstallSquirrel.sh

bajo commented:
@ipa-nhg Another question that came to our mind during testing. Do we still need the openjdk-6-jdk package, or was this only needed for the Festo arm controller?

NadiaHG commented:
I am not sure where is openjdk needed, the arm doesn't need it. We can check it next week after the reconstruction we should clean our repositories

bajo committed:
Introduce ROS_MIRROR variable to easily change the ROS install server

bajo committed:
Reduce output when inserting text into config file

bajo committed:
Escape variables

bajo committed:
change sudoers order

bajo committed:
make sure we can execute this as squirrel with sudo

bajo committed:
Add path for robotino-adduser. 

Otherwise commands executed with sudo are not found in /home/squirrel/squirrel_recommender/setup_robotino

bajo commented:
@ipa-nhg I think now it is finally finished. I ran through the whole install and user adding procedure with this version on tuw-robotino2.

NadiaHG commented:
LGTM :+1:
